Marlin/Marlin_main.cpp

```diff
@@ -5067,46 +5067,65 @@ inline void gcode_G28() {
 
   #if ENABLED(DELTA_AUTO_CALIBRATION)
     /**
-     * G33 - Delta '1-4-7-point' auto calibration (Requires DELTA)
-     * 
-     * Usage:
-     *   G33 <Vn> <Pn> <A> <O> <T>
-     *   
-     *     Vn = verbose level (n=0-2 default 1)
-     *          n=0 dry-run mode: setting + probe results / no calibration
-     *          n=1 settings 
-     *          n=2 setting + probe results 
-     *     Pn = n=-7 -> +7 : n*n probe points
-     *          calibrates height ('1 point'), endstops, and delta radius ('4 points') 
-     *          and tower angles with n > 2 ('7+ points')
-     *          n=1  probes center / sets height only
-     *          n=2  probes center and towers / sets height, endstops and delta radius
-     *          n=3  probes all points: center, towers and opposite towers / sets all
-     *          n>3  probes all points multiple times and averages
-     *     A  = abort 1 point delta height calibration after 1 probe
-     *     O  = use oposite tower points instead of tower points with 4 point calibration
-     *     T  = do not calibrate tower angles with 7+ point calibration
+     * G33 - Delta '1-4-7-point' auto calibration
+     *       Calibrate height, endstops, delta radius, and tower angles.
+     *
+     * Parameters:
+     *
+     *   P  Number of probe points:
+     *
+     *      P1     Probe center and set height only.
+     *      P2     Probe center and towers. Set height, endstops, and delta radius.
+     *      P3     Probe all positions: center, towers and opposite towers. Set all.
+     *      P4-P7  Probe all positions at different locations and average them.
+     *
+     *   A  Abort delta height calibration after 1 probe (P1)
+     *
+     *   O  Use opposite tower points instead of tower points (P2 only)
+     *
+     *   T  Don't calibrate tower angle corrections (for P3-P7)
+     *
+     *   V  Verbose level:
+     *      V0  Dry-run mode. Report settings and probe results. No calibration.
+     *      V1  Report settings
+     *      V2  Report settings and probe results
      */
     inline void gcode_G33() {
 
-      stepper.synchronize();
+      if (axis_unhomed_error(true, true, true)) return;
```

Comment:
I need to call the homing procedure. The calibration procedure needs to home the printer at each iteration otherwise there is no way the printer knows at what position it is after tweaking the end-stops and delta height. It just does not work without homing.